core/templates/components/forms/custom-forms-directives/image-uploader-modal.component.ts

```diff
@@ -0,0 +1,208 @@
+  invalidImageWarningIsShown: boolean = false;
+  windowIsNarrow: boolean = false;
+  invalidTagsAndAttributes: {tags: string[]; attrs: string[]} = {
+    tags: [],
+    attrs: [],
+  };
+  dimensions = {height: 0, width: 0};
+  imageType!: string;
+
+  // 'cropper' is initialized before it is to be used, hence we need to do
+  // non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  cropper!: Cropper;
+  @ViewChild('croppableImage') croppableImageRef!: ElementRef;
+
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    private ngbActiveModal: NgbActiveModal,
+    private windowDimensionService: WindowDimensionsService,
+    private svgSanitizerService: SvgSanitizerService
+  ) {
+    super(ngbActiveModal);
+  }
+
+  private _getAspectRatio(): number {
+    return (
+      parseInt(this.aspectRatio.split(':')[0]) /
+      parseInt(this.aspectRatio.split(':')[1])
+    );
+  }
+
+  initializeCropper(): void {
+    if (this.croppableImageRef) {
+      let imageElement = this.croppableImageRef.nativeElement;
+      if (!this.windowIsNarrow) {
+        this.cropper = new Cropper(imageElement, {
+          minContainerWidth: 500,
+          minContainerHeight: 350,
+          aspectRatio: this._getAspectRatio(),
+        });
+      } else {
+        this.cropper = new Cropper(imageElement, {
+          minContainerWidth: 200,
+          minContainerHeight: 200,
+          aspectRatio: this._getAspectRatio(),
+        });
+      }
+    }
+  }
+
+  onFileChanged(file: Blob): void {
+    this.invalidImageWarningIsShown = false;
+    let reader = new FileReader();
+    reader.onload = e => {
+      this.invalidTagsAndAttributes = {
+        tags: [],
+        attrs: [],
+      };
+      let imageData = (e.target as FileReader).result as string;
+      if (this.svgSanitizerService.isBase64Svg(imageData)) {
+        this.invalidTagsAndAttributes =
+          this.svgSanitizerService.getInvalidSvgTagsAndAttrsFromDataUri(
+            imageData
+          );
+        this.thumbnailImageDataUrl =
+          this.svgSanitizerService.removeAllInvalidTagsAndAttributes(imageData);
+        this.uploadedImage =
+          this.svgSanitizerService.getTrustedSvgResourceUrl(imageData);
+      }
+      if (!this.uploadedImage) {
+        this.uploadedImage = decodeURIComponent(
+          (e.target as FileReader).result as string
+        );
+      }
+
+      const img = new Image();
+      img.onload = () => {
+        this.dimensions = {height: img.height, width: img.width};
+
+        try {
+          this.changeDetectorRef.detectChanges();
+        } catch (viewDestroyedError) {
+          // This try catch block handles the following error in FE tests:
+          // ViewDestroyedError:
+          //   Attempt to use a destroyed view: detectChanges thrown.
+          // No further action is needed.
+        }
+        if (this.imageName !== 'Thumbnail') {
```

Comment:
I've seen too many of "imageName !== 'Thumbnail'" in this PR, this should be a separate function

Reply:
Done. created isThumbnail() function.

Reply:
You forgot to replace this line

Reply:
Done.